### DISCUSSION TITLE
Basic questions on metabolic modelling of microbial communities

### DISCUSSION THREAD

**Discussion by wintermute221** [micom-dev/micom#40] [Q&A] [answered]
Hi!!

The lab I work in mainly focuses on using experimental methods to investigate the human gut microbial communities and now we would like to try some metabolic modelling tools and see if they can be applied in our research. We found MICOM is a useful tool to explore the co-existence of microbes. 

The problem is, neither my supervisor nor my peers have any experience in computational approaches. I only have some experience with Python and genome-scale metabolic models. So I have some very basic questions regarding metabolic modelling of communities and any ideas/advice will help a lot!

First question:
        How does community optimisation packages do optimisations? Because apparently the two strains both will try their best to grow, then how will the package decide whose objective function it will optimize? Or will the objective function will be a mix of the two microbes'?

Second question:
        In a simulation work if one strain grows slowly and the other grows very fast, can it really reflect the practical experiment?

Also appreciate it if there are any books/literatures suggested to find the answers!

**Comment by cdiener** [comment]
Hi!

Glad you're finding MICOM helpful. Getting into metabolic modeling can sometimes feel daunting but can be helpful when coming from an experimental background. In particular, if you want to identify the metabolic interactions (trophism, competition etc.) that drive co-existence. Okay, let's get to your questions.

1) Brilliant question and in fact the question that motivated the creation of MICOM. As you said, there are two types of biomass/growth one has to consider: taxa-specific growth and the total biomass produced by all taxa in unison. MICOM considers both by ensuring a high but not necessarily maximum total biomass production and but also distributing growth across individual taxa. Empirically this is meant to model the tradeoff between community-wide and taxa-specific growth maximization by finding a good intermediate point. The resources further down explain that in more detail.

2) So the honest answer is: it can but does not always have to. The better your metabolic model reflects the strain or taxonomic group you're modeling and the more defined your growth medium the better the prediction. We did try to investigate that a bit in the original publication by comparing the predicted growth rates from MICOM to replication rates inferred from metagenomic data:

![](https://journals.asm.org/cms/10.1128/mSystems.00606-19/asset/0169e61f-76b1-45fa-95d3-a92b8830d50b/assets/graphic/msystems.00606-19-f0001.jpeg)

Figure C shows the correlation between the two and as you see it does a better job than just optimizing community-wide growth (which is what was commonly done before). On average you do get a decent correlation but it is definitely not perfect. We are looking at some co-culture experiments as well and you can get very good agreement but that depends on how well you have characterized the growth medium and the metabolism of the contained metabolites in the strain-level models. 

Regarding resources, the [tutorial section](https://micom-dev.github.io/micom/tutorials.html) of the MICOM docs lists a course that explains the optimizations in MICOM in detail and provide a practical walkthrough.

**Reply by wintermute221** [reply]
Thank you for your answers very much. They have provided very useful information!

Regarding your first answer, I just wonder how to find a reasonable intermediate point when modelling the tradeoff between community-wide and taxa-specific growth maximization? As shown above, Figure C shows different trade-off values ranging from 0.1 to 1.0 so is it suggesting that if we want to find an intermediate point we need to try different trade-off values on the two strains? Or perhaps some experimental data is needed in this step?

Besides, I've got some further questions related to Figure C. I just noticed the first seven box plots (trade-off values from 0.1 to 0.7) are very similar to each other. I don't know if I understand it wrongly but I just wonder how to decide which trade-off value to use given their values are very close to each other? Because from my understanding, the trade-off value will decide the abundance of each strain in the community when conducting the community model and hence will eventually influence the objective function.

Many thanks!

**Reply by cdiener** [reply]
You probably want to maintain the largest tradeoff that still allows growth for the majority of the community. So you would choose the largest value that allows for the majority of taxa to grow on average. The docs [have a bit more info on that](https://micom-dev.github.io/micom/viz.html#Choosing-a-tradeoff-value) and it is pretty easy and can be done with two commands for most cases. 

For two strain models, we find that the tradeoff is usually not very important since there isn't enough competition and the flux space is not as large. You can use the same method as above though: simulate all pairwise cultures, then choose the largest tradeoff where both strains grow in the majority of samples (the points will be very discrete here since either none growth - 0%, one growth - 50%, or both grow - 100%). For instance, this is what this looks like for simulations of about 4000 cocultures, from isolates of the same species:

![image](https://user-images.githubusercontent.com/7153935/122449795-1cf93600-cf5b-11eb-8009-2a4b88b152ce.png)

As you see a tradeoff of 1 works just fine there. Using experimental data would be great and is always the preferred option if available. For instance, run a small set of cocultures and see at which tradeoff you get the best correlation with the actual growth rate. Then use that to predict the growth rates of additional cocultures and see how well it works.

For the second question that is the nature of the tradeoff parameter. Lower tradeoff distributes the growth rates more across individual taxa. If a high tradeoff does well a lower will do even better. So It will always be approximately monotonous. Once you pass the "elbow" lowering the threshold even further will just lower the overall growth rates but their relations remain the same. So 0.6 and 0.3 growth rates for two taxa with tradeoff=0.8 will become 0.3 and 0.15 with a tradeoff=0.4 here. There is an argument in the paper why that happens in the SI, but I think the take-home message is to choose the largest tradeoff that gives sensible results (most taxa growing, agreement with initial experimental data).

**Reply by wintermute221** [reply]
Thanks very much. I don't have any more questions currently. I will look through the docs and see if I've have any questions then.

**Reply by wintermute221** [reply]
Hi! Some more interesting questions just came to my mind. Please correct me if there's anything wrong.

1. If we say we put individual A and individual B in medium 1 and find they can grow together in the lab. And then we change the medium to medium 2 and find that individual A stop individual B from growing. So how would the simulation reflect the reality? Because from my basic understanding I assume the simulation would work only if the two strains both grow in the majority of samples.

2. I can see the general idea is to set the community growth rate to a certain fraction of the maximum community growth rate and maximise the individual growth rates. So if there're more than two strains in the community how would you decide which individual growth rate you would maximise?

**Reply by cdiener** [reply]
Let's see :thinking: 

1. Yes, if you estimate the tradeoff that should be done in a rich medium where you expect all taxa to grow. If your data set is a single co-culture you would probably not estimate the tradeoff parameters since that are not enough data points. Here you would choose one *a priori* and see if the models reproduce the behavior well enough. For co-cultures a tradeoff of 1 or 0.9 is probably sufficient as argued before. You can just plugin different media for a model with MICOM and check if that would lead to B stopping to grow or growing very slow compared to the other medium.

2. Yep, that would be very hard and is considered a multi-level optimization which is fairly hard to solve even for 2 microbes. MICOM goes around that by enforcing the fraction and then distributing the growth rates as evenly across the other taxa as possible, which is a problem that can be solved much faster and that has the added benefit of having a unique solution (multi-level optimization can have several optimal solutions). The paper has some more background on why that is related to optimizing the individual growth rates.

**Reply by wintermute221** [reply]
Thanks for the answers. There're only a few points I'd love to double check. 

1. I'm not sure I can understand it 100%. Are you suggesting that we just optimise the growth rate of A if A stops B from growing and then compare the results with those using different media? As for this case, I'm not sure if there's a need to set up the tradeoff value to 1 or 0.9 since I assume 'tradeoff' only exists when all the taxa can grow and if only one taxa can grow in the medium we could just optimise its growth rate.

2. From your description, am I understanding it correctly that the package will manually set the growth rate of each taxa the same? If so, I just wonder if it could reflect the reality since I assume in a community different taxa will have different growth rates from each other.

**Comment by cdiener** [comment, marked as answer]
Hm, I think I did not explain that well. Sorry.

1. Nope, my suggestion was to choose a fixed tradeoff of 1.0 or 0.9 and then just simulate with that and see if it reproduces the experiment. Having a tradeoff does not mean all taxa can grow, it will try to find a solution where that is possible but if the nutritional requirements are not met an organism might still be blocked from growth. And you do need the regularization to get a unique solution, otherwise, there may be infinitely many solutions for growth rates for A and B, some of those inhibiting growth of A and some that don't. 

2. No, that is not what MICOM will do. It does distribute the growth rate across the community but is considering all constraints and by still achieving the required total biomass. Think of it like a tug of war where one side tries to distribute growth rates according to the abundance of the taxa and the other side tries to get the entire community to grow as fast as possible. This will usually not result in the same growth rate for all taxa, except in the case where all taxa are exactly the same strain or at least metabolically identical. MICOM will predict different growth rates for different taxa as long as the tradeoff parameter is chosen sufficiently large (which is why we recommend using the largest one allowing enough taxa to grow). There are some frameworks that do assume identical growth rates for all organisms though (for instance [SteadyCom](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005539)).

**Reply by wintermute221** [reply]
Thank you!! Sorry for my late reply. I was insanely busy with my exams and just had all them done. I'll think over it and let you know any further questions. 😄
